server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/hover/ParagraphHoverProvider.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package org.eclipse.lsp.cobol.service.delegates.hover;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import lombok.NonNull;
+import org.eclipse.lsp.cobol.common.AnalysisResult;
+import org.eclipse.lsp.cobol.common.model.DefinedAndUsedStructure;
+import org.eclipse.lsp.cobol.common.model.Describable;
+import org.eclipse.lsp.cobol.common.model.tree.*;
+import org.eclipse.lsp.cobol.common.utils.RangeUtils;
+import org.eclipse.lsp.cobol.lsp.SourceUnitGraph;
+import org.eclipse.lsp.cobol.lsp.handlers.text.HoverHandler;
+import org.eclipse.lsp.cobol.service.CobolDocumentModel;
+import org.eclipse.lsp4j.*;
+
+/** The class provides hover information for Procedures and Sections. */
+public class ParagraphHoverProvider implements HoverProvider {
+  private static final int MAX_HOVER_LINE_COUNT = 20;
+
+  /**
+   * @param document - document model that contains a semantic context
+   * @param position - cursor position
+   * @param documentGraph - workspace doc graph
+   * @return
+   */
+  @Nullable
+  @Override
+  public Hover getHover(
+      @Nullable CobolDocumentModel document,
+      @NonNull TextDocumentPositionParams position,
+      SourceUnitGraph documentGraph) {
+    Optional<AnalysisResult> analysisResult =
+        Optional.ofNullable(document).map(CobolDocumentModel::getAnalysisResult);
+    if (!analysisResult.isPresent()) return null;
+    Optional<RootNode> rootNode =
+        Optional.of(document)
+            .map(CobolDocumentModel::getAnalysisResult)
+            .map(AnalysisResult::getRootNode);
+    Optional<Node> nodeOpts =
+        rootNode.flatMap(
+            root ->
+                RangeUtils.findNodeByPosition(
+                    root, position.getTextDocument().getUri(), position.getPosition()));
+    if (!nodeOpts.isPresent()) return null;
+    Node node = nodeOpts.get();
+    if (!(node instanceof DefinedAndUsedStructure)) return null;
+    List<Location> definitions = ((DefinedAndUsedStructure) node).getDefinitions();
+    if (definitions == null || definitions.isEmpty()) return null;
+
+    List<Node> nodeStream =
+        rootNode
+            .get()
+            .getDepthFirstStream()
+            .filter(n -> definitions.contains(n.getLocality().toLocation()))
+            .collect(Collectors.toList());
+
+    Stream<Hover> hoverStream =
+        getHoverStream(nodeStream, (DefinedAndUsedStructure) node, document.getLanguageId());
+    if (nodeStream.size() > 1) {
+      hoverStream =
+          Stream.concat(
+              hoverStream,
+              Stream.of(
+                  new Hover(
+                      new MarkupContent(
+                          "text", "___NOTE: other versions exist due to replac(e/ing)___"))));
```

Comment:
The triple `___` looks very distracting to me.

```suggestion
                          "text", "_NOTE: other versions exist due to replacing_"))));
```